mono/metadata/threads-types.h

```diff
@@ -274,12 +274,18 @@ void mono_thread_join (gpointer tid);
 
 void ves_icall_System_Threading_Thread_GetStackTraces (MonoArray **out_threads, MonoArray **out_stack_traces);
 
-MONO_API gpointer
+MONO_RT_EXTERNAL_ONLY MONO_API gpointer
```

Comment:
They are in public API as lambdageek pointed out, just the two attach/detach functions as I understand.

Reply:
They are not in the public API since they are not declared in a header that is public. And they do not have any raw object pointers. So they do not qualify.

Reply:
Oh you mean "public API" vs. "used by xamarin", so the second is very bounded and fixable/changable? I should update them? It turns out they duplicate all the GC state transition macros. Perhaps that duplicate can/should be eliminated?

Reply:
I think I should have foo be unchanged be used external and JIT;
foo_internal is the new one; or foo_with_func but that is maybe too specific
foo_common is underlying implementation
So line damage in JIT removed and some other line damage maybe undone.
Just the runtime's use of the macros will improve.

Or maybe put time into fixing backtrace on linux, to use libelf??? And then scrap all this??